### PR TITLE
Add send_mana and prepare_send_mana

### LINKED ISSUE #2028: Add send-mana to rust core and CLI
With option to set a storage deposit return unlock condition to get the amount back

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fixes https://github.com/iotaledger/iota-sdk/issues/2028

### CODE REVIEW

File: sdk/src/wallet/operations/transaction/prepare_output.rs [hunk 1]
```diff
@@ -113,7 +113,7 @@ where
         let min_amount_basic_output =
             BasicOutput::minimum_amount(&Address::from(Ed25519Address::null()), storage_score_params);
 
-        let min_required_storage_deposit = first_output.minimum_amount(storage_score_params);
+        let min_required_storage_deposit = first_output.amount();
```
Comment (thibault-martinez) [inline sdk/src/wallet/operations/transaction/prepare_output.rs:116]: Equivalent, right?

Comment (Thoralf-M) [reply sdk/src/wallet/operations/transaction/prepare_output.rs:116]: Yes, if the calculation there is not different inside :trollface:

File: sdk/src/wallet/operations/transaction/high_level/send_mana.rs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        log::debug!("[TRANSACTION] prepare_send_mana");
+        let return_strategy = params.return_strategy.unwrap_or_default();
+        let storage_score_params = self.client().get_storage_score_parameters().await?;
+
+        let mut output_builder = BasicOutputBuilder::new_with_minimum_amount(storage_score_params)
+            .with_mana(params.mana)
+            .add_unlock_condition(AddressUnlockCondition::new(params.address));
+
+        match return_strategy {
+            ReturnStrategy::Return => {
```
Comment (Thoralf-M) [inline sdk/src/wallet/operations/transaction/high_level/send_mana.rs:77]: Clippy not complaining to use `if return_strategy == ReturnStrategy::Return{}` instead?

Comment (thibault-martinez) [reply sdk/src/wallet/operations/transaction/high_level/send_mana.rs:77]: No, I'm using a match so it fails when/if we add a new strategy, wouldn't fail with the if. I guess Clippy understands that.

File: sdk/src/wallet/operations/transaction/prepare_output.rs (hunk 1 above)
Comment (thibault-martinez) [reply sdk/src/wallet/operations/transaction/prepare_output.rs:116]: Don't think it is

**Review by Thoralf-M — DISMISSED** [review]

File: cli/src/wallet_cli/mod.rs [hunk 3]
```diff
@@ -272,6 +272,16 @@ pub enum WalletCommand {
         #[arg(long, default_value_t = false)]
         allow_micro_amount: bool,
     },
+    /// Send mana.
+    SendMana {
+        /// Address to send mana to, e.g. rms1qztwng6cty8cfm42nzvq099ev7udhrnk0rw8jt8vttf9kpqnxhpsx869vr3.
```
Comment (DaughterOfMars) [inline cli/src/wallet_cli/mod.rs]:
```suggestion
        /// Recipient address, e.g. rms1qztwng6cty8cfm42nzvq099ev7udhrnk0rw8jt8vttf9kpqnxhpsx869vr3.
```

**Review by DaughterOfMars — COMMENTED** [review]
Bindings?

**Comment by thibault-martinez** [comment]
> Bindings?

https://github.com/iotaledger/iota-sdk/issues/1990

**Review by DaughterOfMars — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**